docs/user_guide/02b_query_engine.md

```diff
@@ -1 +1,294 @@
+
+# Wrong implementation
+gender = node.attribute("gender")
+gender.lowercase().trim()
+gender.equal_to("m")
+
+# Correct implementation
+gender = node.attribute("gender")
+gender.lowercase()
+gender.trim()
+gender.equal_to("m")
+```
+
+Nor do the ones that compare operands to other operands, since their `Return` value is also null.
+
+```python
+# Wrong implementation
+gender = node.attribute("gender")
+gender.equal_to("M").not_equal_to("F")
+
+# Correct implementation
+gender = node.attribute("gender")
+gender.equal_to("M")
+gender.not_equal_to("F")
+```
+
+:::
+
+As you can see, the query engine is highly useful for finding nodes that fulfill different criteria in a highly optimized way, using descriptive programming as a precise tool. We can use previously defined queries too, and also use the [`neighbors()`](medmodels.medrecord.querying.NodeOperand.neighbors){target="_blank"} method to query also through the nodes that are neighbors to those nodes.
+
+For instance, in the following example we are selecting the nodes that have the following characteristics
+
+- Are in group `patient`.
+- Their node index contains the string _"pat"_
+- Their attribute `age` is greater than 30, and their attribute `gender` is equal to _"M"_.
+- They are  connected to  nodes which attribute `description` contains the word _"fentanyl"_  in either upper or lowercase.
+
+```{exec-literalinclude} scripts/02b_query_engine.py
+---
+language: python
+setup-lines: 1-4, 15-24
+lines: 46-55
+---
+```
+
+:::{dropdown} Methods used in the snippet
+
+- [`neighbors()`](medmodels.medrecord.querying.NodeOperand.neighbors){target="_blank"} : Returns a [`NodeOperand()`](medmodels.medrecord.querying.NodeOperand){target="_blank"} to query the neighbors of those nodes.
+- [`attribute()`](medmodels.medrecord.querying.NodeOperand.attribute){target="_blank"} : Returns a [`MultipleValuesOperand()`](medmodels.medrecord.querying.MultipleValuesOperand){target="_blank"} to query on the values of the nodes for that attribute.
+- [`lowercase()`](medmodels.medrecord.querying.MultipleValuesOperand.lowercase){target="_blank"} : Converts the values that are strings to lowercase.
+- [`contains()`](medmodels.medrecord.querying.NodeIndexOperand.contains){target="_blank"} : Query node indices containing that argument.
+- [`select_nodes()`](medmodels.medrecord.medrecord.MedRecord.select_nodes){target="_blank"} : Select nodes that match that query.
+
+:::
+
+## Edge Queries
+
+Edge operands provide a way to define and filter relationships between nodes in a [`MedRecord`](medmodels.medrecord.medrecord.MedRecord){target="_blank"}. By using edge operands, you can create queries that target specific connections based on attributes like time, duration, or custom properties. This section explores how to leverage edge operands to gain insights from the relationships in your data. For that, the [`EdgeOperand`](medmodels.medrecord.querying.EdgeOperand){target="_blank"} querying class is used.
```

Comment:
> Edge operands provide a way to define and filter relationships between nodes in a [...]

should be edges not nodes.

---

> By using edge operands, you can create queries that target specific connections based on attributes like time, duration, or custom properties.

Sounds very limiting as edges has basically the same functionalities in this regard (and also the others) as the `NodeOperand`

---

> This section explores how to ✨leverage✨ edge operands to ✨gain insights✨ from [...]

also kinda chatgpt-ey